src/main/java/com/tranhiep/listener/RequestEventListener.java

```diff
@@ -0,0 +1,40 @@
+package com.tranhiep.listener;
+
+import com.tranhiep.util.HibernateUtils;
+
+import javax.persistence.EntityManager;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletRequestEvent;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.annotation.WebListener;
+import javax.servlet.http.HttpServletRequest;
+
+@WebListener
+public class RequestEventListener implements ServletRequestListener {
+
+    @Override
+    public void requestInitialized(ServletRequestEvent sre) {
+        HttpServletRequest req = (HttpServletRequest) sre.getServletRequest();
+
+        String reqURI = req.getRequestURI();
+
+        // filter some static request
+        if(!reqURI.endsWith(".js") && !reqURI.endsWith(".css") && !reqURI.endsWith(".png")){
+            EntityManager entityManager = HibernateUtils.getEntityManager();
+            req.setAttribute("entityManager", entityManager);
```

Comment:
`req.setAttribute("entityManager", entityManager);`
Please explain why need to set `entityManager` on every request?

Reply:
dạ . Entitymanager sử dụng để tương tác với hibernate 
 - em sử dụng pattern Entitymanager-per-Request nên tạo entitymanager cho mỗi request rồi inject vào service, service inject vào dto để tương tác vs database.